### PR TITLE
Enable garbage collection for NATS-Streaming server channels

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Enable garbage collection for NATS-Streaming server channels.

**Related issue(s)**
#1233

### CODE REVIEW

**Comment by marcobebway** [comment]
@montaro, yes 48h is discussed and mentioned here: #1233.

**Comment by marcobebway** [comment]
* Verified that the max_inactivity is configured in nats-steraming:
```
$ kubectl logs -f -n kyma-system core-nats-streaming-0 -c nats-streaming

[1] 2018/10/17 21:30:42.853873 [INF] STREAM: Starting nats-streaming-server[kyma-core-nats-streaming] version 0.11.0
[1] 2018/10/17 21:30:42.854426 [INF] STREAM: ServerID: NI8f2IQ0lthEDottHdkDGP
[1] 2018/10/17 21:30:42.854434 [INF] STREAM: Go version: go1.11
[1] 2018/10/17 21:30:42.855177 [INF] Starting nats-server version 1.3.0
[1] 2018/10/17 21:30:42.855186 [DBG] Go build version go1.11
[1] 2018/10/17 21:30:42.855189 [INF] Git commit [not set]
[1] 2018/10/17 21:30:42.855321 [INF] Starting http monitor on 0.0.0.0:8222
[1] 2018/10/17 21:30:42.855361 [INF] Listening for client connections on 0.0.0.0:4222
[1] 2018/10/17 21:30:42.855366 [DBG] Server id is cmvbDlm4zNj2Avi19X5ftT
[1] 2018/10/17 21:30:42.855368 [INF] Server is ready
[1] 2018/10/17 21:30:42.880769 [DBG] 127.0.0.1:34372 - cid:1 - Client connection created
[1] 2018/10/17 21:30:42.881848 [DBG] 127.0.0.1:34374 - cid:2 - Client connection created
[1] 2018/10/17 21:30:42.882591 [DBG] 127.0.0.1:34376 - cid:3 - Client connection created
[1] 2018/10/17 21:30:42.882981 [INF] STREAM: Recovering the state...
[1] 2018/10/17 21:30:42.883235 [INF] STREAM: No recovered state
[1] 2018/10/17 21:30:43.136603 [DBG] STREAM: Did not detect another server instance
[1] 2018/10/17 21:30:43.136685 [DBG] STREAM: Discover subject:           _STAN.discover.kyma-core-nats-streaming
[1] 2018/10/17 21:30:43.136698 [DBG] STREAM: Publish subject:            _STAN.pub.NI8f2IQ0lthEDottHdkDJa.>
[1] 2018/10/17 21:30:43.136701 [DBG] STREAM: Subscribe subject:          _STAN.sub.NI8f2IQ0lthEDottHdkDJa
[1] 2018/10/17 21:30:43.136702 [DBG] STREAM: Subscription Close subject: _STAN.subclose.NI8f2IQ0lthEDottHdkDJa
[1] 2018/10/17 21:30:43.136704 [DBG] STREAM: Unsubscribe subject:        _STAN.unsub.NI8f2IQ0lthEDottHdkDJa
[1] 2018/10/17 21:30:43.136706 [DBG] STREAM: Close subject:              _STAN.close.NI8f2IQ0lthEDottHdkDJa
[1] 2018/10/17 21:30:43.136885 [INF] STREAM: Message store is FILE
[1] 2018/10/17 21:30:43.136890 [INF] STREAM: Store location: /var/lib/nats-streaming/core-nats-streaming/core-nats-streaming-0
[1] 2018/10/17 21:30:43.136915 [INF] STREAM: ---------- Store Limits ----------
[1] 2018/10/17 21:30:43.136917 [INF] STREAM: Channels:                  100 *
[1] 2018/10/17 21:30:43.136918 [INF] STREAM: --------- Channels Limits --------
[1] 2018/10/17 21:30:43.136920 [INF] STREAM:   Subscriptions:          1000 *
[1] 2018/10/17 21:30:43.136921 [INF] STREAM:   Messages     :       1000000 *
[1] 2018/10/17 21:30:43.136923 [INF] STREAM:   Bytes        :     976.56 MB *
[1] 2018/10/17 21:30:43.136924 [INF] STREAM:   Age          :       24h0m0s
[1] 2018/10/17 21:30:43.136925 [INF] STREAM:   Inactivity   :       48h0m0s
[1] 2018/10/17 21:30:43.136927 [INF] STREAM: ----------------------------------
```

* Verified that event-bus examples are working correctly.

File: resources/core/charts/event-bus/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Configure these options for each business requirement:
 | | `global.persistence.size` | The size of the storage volume. |
 | | `global.persistence.maxAge`| The maximum period of time for storing an event (`0` for unlimited). |
 | |`global.natsStreaming.resources`| Refer to Kubernetes resource requests and limits for details. |
+| |`global.natsStreaming.channel.maxInactivity`| Max inactivity (no new event, no subscription) after which a channel can be garbage collected (`0` for unlimited) |
```
Comment (bszwarc) [inline resources/core/charts/event-bus/README.md]: `The maximum inactivity period (without any new event or subscription) after which a channel can be garbage collected (`0` for unlimited).`

File: resources/core/charts/event-bus/README.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Configure these options for each business requirement:
 | | `global.persistence.size` | The size of the storage volume. |
 | | `global.persistence.maxAge`| The maximum period of time for storing an event (`0` for unlimited). |
 | |`global.natsStreaming.resources`| Refer to Kubernetes resource requests and limits for details. |
+| |`global.natsStreaming.channel.maxInactivity`| The maximum inactivity period (without any new event or subscription) after which a channel can be garbage collected (`0` for unlimited). |
```
Comment (bszwarc) [inline resources/core/charts/event-bus/README.md]: event > Event

**Pull request merged**